java-checks/src/test/files/checks/StringConcatenationInLoopCheck.java

```diff
@@ -58,4 +58,17 @@ static MyObject newInstance() {
     }
   }
 
+  class M {
+    String f;
+
+    void foo(M m) {
+      while (true) {
+        (this).f += "sdasd"; // Noncompliant
+
+        (m = this).f += "asdasd"; // not supported... what to do with this?
+        ((M) m).f += "asdas"; // not supported... what to do with this?
+      }
+    }
+  }
+
 }
```

Comment:
Should we move this file to the non-compiling sources folder of `java-checks-test-sources` ?

Reply:
Indeed, we should. no reason to not make it clean.